internal/cmd/common.go

```diff
@@ -127,3 +129,62 @@ func handleTestAbortSignals(gs *state.GlobalState, gracefulStopHandler, onHardSt
 		gs.SignalStop(sigC)
 	}
 }
+
+func resolveStackSlugToID(gs *state.GlobalState, jsonRawConf json.RawMessage, token, slug string) (int64, error) {
+	slug = stripGrafanaNetSuffix(slug)
+	stacks, err := getStacks(gs, jsonRawConf, token)
+	if err != nil {
+		return 0, err
+	}
+	id, ok := stacks[slug]
+	if !ok {
+		return 0, fmt.Errorf("stack slug %q not found in your Grafana Cloud account", slug)
+	}
+	return int64(id), nil
+}
+
+func stripGrafanaNetSuffix(s string) string {
+	const suffix = ".grafana.net"
+	if len(s) > len(suffix) && s[len(s)-len(suffix):] == suffix {
+		return s[:len(s)-len(suffix)]
+	}
+	return s
+}
+
+func resolveDefaultProjectID(
+	gs *state.GlobalState,
+	apiClient *cloudapi.Client,
+	jsonRawConf json.RawMessage,
+	token string,
+	stackSlug null.String,
+	stackID *null.Int,
+	projectID *int64,
+) (int64, error) {
+	if *projectID != 0 {
+		return *projectID, nil
+	}
+
+	if stackID.Int64 == 0 {
+		if stackSlug.Valid && stackSlug.String != "" {
+			id, err := resolveStackSlugToID(gs, jsonRawConf, token, stackSlug.String)
+			if err != nil {
+				return 0, fmt.Errorf("could not resolve stack slug %q to stack ID: %w", stackSlug.String, err)
+			}
+			*stackID = null.IntFrom(id)
+		} else {
+			// TODO: do this in the future? But it is a breaking change...
+			// gs.Logger.Error("please specify a projectID in your test or use `k6 cloud login` to set up a default stack")
+			// return 0, fmt.Errorf("no projectID specified and no default stack set")
```

Comment:
We can do two things here:

1. Add a warning and make the user aware of the future changes without breaking the current logic
2. We can start having feature flags with breaking changes, we might also have some `next` feature flag, where we enable all the feature in the shape of `v2`. So, the users that don't care about breaking changes can benefit of the new features.